File: code/SoundPlayground/SoundPlayground/Parser/AbstractSyntaxTree/ContextModifiers/LengthModifierNode.cs
using System.Collections.Generic;
using SoundPlayground.Core;

namespace SoundPlayground.Parser.AbstractSyntaxTree.ContextModifiers {
    public class LengthModifierNode : ContextModifierNode {
        public float Length { get; set; }

        public LengthModifierNode ( float length ) {
            Length = length;
        }

        public override void Modify ( Context context ) {
            context.Value = Length;
        }
    }
}

File: code/SoundPlaygroundPy/parser/abstract_syntax_tree/context_modifiers/length_modifier_node.py
from .context_modifier_node import ContextModifierNode
from core.events import ContextChangeEvent
from core import Context, Voice

class LengthModifierNode( ContextModifierNode ):
    def __init__ ( self, length, position : (int, int) = None ):
        super().__init__( position )

        self.length = length
        
    def apply ( self, voice : Voice ):
        voice.value = self.length

    def modify ( self, context : Context ):
        context.voice = context.voice.clone( value = self.length )

        yield ContextChangeEvent( context.cursor, "length", context.voice.value )
